modelator/src/model/language/tla/mod.rs

```diff
@@ -168,23 +168,78 @@ impl Tla {
     }
 }
 
+/// Returns a list of operator names
+fn extract_operator_names(tla_file_contents: &str) -> Result<Vec<String>, Error> {
+    let cnt_operators = tla_file_contents.match_indices("==").count();
+    lazy_static! {
+        // Match '<identifier><whitespace>=='
+        static ref RE: Regex = Regex::new(r"([a-zA-Z0-9_]*)\s*==").unwrap();
+    }
+    let ret: Vec<String> = RE
+        .captures_iter(tla_file_contents)
+        .filter_map(|caps| caps.get(0))
+        .map(|m| m.as_str().to_owned())
+        .map(|s| s.trim_end_matches('=').trim().to_owned())
+        .collect();
+
+    match ret.len() == cnt_operators {
+        true => Ok(ret),
+        false => Err(Error::TlaOperatorNameParseError(
+            tla_file_contents.to_owned(),
+        )),
+    }
+}
+
+/// Scan the contents of the tla file to try to find an operator named
+/// '<test_name>View'. If none is found then use an operator named 'View', if found.
+/// If no operator is found then returns None.
+fn extract_view_operator(
+    test_name: &str,
+    tla_file_contents: &str,
+) -> Result<Option<String>, Error> {
+    let operators = extract_operator_names(tla_file_contents)?;
+    Ok(operators.iter().fold(None, |acc, s| -> Option<String> {
+        // If the operator is a test specific view then use it
+        if *s == format!("{}View", test_name) {
```

Comment:
merge these `if` blocks together as `if-then-else-if-then-else-if-then-else-...` and remove `return` keywords.

Reply:
Nice rusty style point, ty!